pkg/s3/bucket.go

```diff
@@ -146,22 +151,81 @@ func CreateBucketTaggingInput(bucketname string, tags map[string]string) *s3.Put
 	return putInput
 }
 
+// ClearBucketTags wipes all existing tags from a bucket so that velero-specific
+// tags can be applied to the bucket instead.
 func ClearBucketTags(s3Client *s3.S3, bucketName string) (err error) {
 	deleteInput := &s3.DeleteBucketTaggingInput{Bucket: aws.String(bucketName)}
 	_, err = s3Client.DeleteBucketTagging(deleteInput)
 	return err
 }
 
-func TagBucket(s3Client *s3.S3, bucketName string, backUpLocation string) error {
+// TagBucket adds tags to an S3 bucket. The tags are used to indicate that velero backups
+// are stored in the bucket, and to identify the associated cluster.
+func TagBucket(s3Client *s3.S3, bucketName string, backUpLocation string, infraName string) error {
 	err := ClearBucketTags(s3Client, bucketName)
```

Comment:
Hmmm. I didn't realize we actually wiped tags every time we call this. This isn't really impotent, is it?

Is this intended? What were your thoughts behind this, @c-e-brumm?

Reply:
The idea was that if we always deleted the tags and reset the tags they will always be the tags we want. This way a new tag can be added or a tag can be changed in one place in the code and it will always take effect on existing and new clusters. We won't need additional logic to validate tags if they get reset every time. Heavy handed but simple was the thought.

Reply:
I gotcha. There are some potential issues I see with this in the future, so we may want to revisit at some point in the future to make this code more robust.